braindecode/models/eegnet.py

```diff
@@ -38,32 +43,27 @@ class EEGNetv4(EEGModuleMixin, nn.Sequential):
         Length of the depthwise convolution kernel in the separable convolution.
     pool1_kernel_size : int, default=4
         Kernel size of the first pooling layer.
-    pool1_stride_size : int, default=4
-        Stride size of the first pooling layer.
     pool2_kernel_size : int, default=8
         Kernel size of the second pooling layer.
-    pool2_stride_size : int, default=8
-        Stride size of the second pooling layer.
     kernel_length : int, default=64
         Length of the temporal convolution kernel.
     conv_spatial_max_norm : float, default=1
-        Max norm constraint for the spatial convolution layer.
+        Maximum norm constraint for the spatial (depthwise) convolution.
     activation : nn.Module, default=nn.ELU
-        Activation function to apply. Should be a PyTorch activation module like
-        ``nn.ReLU`` or ``nn.ELU`` after the batch normalization layer.
+        Non-linear activation function to be used in the layers.
     batch_norm_momentum : float, default=0.01
-        Momentum for the batch normalization layers.
+        Momentum for instance normalization in batch norm layers.
     batch_norm_affine : bool, default=True
-        Whether to include learnable affine parameters in batch normalization layers.
+        If True, batch norm has learnable affine parameters.
     batch_norm_eps : float, default=1e-3
-        Epsilon value for batch normalization layers.
+        Epsilon for numeric stability in batch norm layers.
     drop_prob : float, default=0.25
-        Dropout probability after the second conv block and before the last layer.
-
-    Notes
-    -----
-    This implementation is not guaranteed to be correct, has not been checked
-    by original authors, only reimplemented from the paper description.
+        Dropout probability.
+    final_layer_linear : bool, default=False
+        If ``False``, uses a convolution-based classification layer. If ``True``,
+        apply a flattened linear layer as the final classification step.
```

Comment:
```suggestion
    final_layer_with_constraint : bool, default=False
        If ``False``, uses a convolution-based classification layer. If ``True``,
        apply a flattened linear layer with constraint on the weights norm as the final classification step.
```
The fact that it's linear vs. convolutional is not super important, the main difference is the constraint. So I think the name should reflect that